### PR TITLE
add create helper functions for custom cards

### PR DESCRIPTION
**Pull request by iantrich**
## Proposed change

Exposing the create helpers functions so that custom cards can use them. Will add this to the dev docs as they are added.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

Simple custom card implementation to use it
```ts
/* eslint-disable @typescript-eslint/no-explicit-any */
import { LitElement, html, customElement, property, CSSResult, TemplateResult, css, PropertyValues } from 'lit-element';
import { HomeAssistant, hasConfigOrEntityChanged } from 'custom-card-helpers';
import deepClone from 'deep-clone-simple';

import { CustomCardConfig } from './types';
import { CARD_VERSION } from './const';

import { localize } from './localize/localize';

/* eslint no-console: 0 */
console.info(
  `%c  custom-card \n%c  ${localize('common.version')} ${CARD_VERSION}    `,
  'color: orange; font-weight: bold; background: black',
  'color: white; font-weight: bold; background: dimgray',
);

@customElement('custom-card')
export class CustomCard extends LitElement {
  @property() public hass?: HomeAssistant;
  @property() private _config?: CustomCardConfig;
  @property() private _helpers?: any;

  public setConfig(config: CustomCardConfig): void {
    if (!config) {
      throw new Error(localize('common.invalid_configuration'));
    }

    this._config = {
      ...config,
    };

    this.loadCardHelpers();
  }

  protected shouldUpdate(changedProps: PropertyValues): boolean {
    return hasConfigOrEntityChanged(this, changedProps, false);
  }

  protected render(): TemplateResult | void {
    if (!this._config || !this.hass || !this._helpers) {
      return html``;
    }

    const config = deepClone(this._config);
    delete config['type'];
    const element = this._helpers.createRowElement(config);

    if (this.hass) {
      element.hass = this.hass;
    }

    return html`
      <div>${element}</div>
    `;
  }

  private async loadCardHelpers(): Promise<void> {
    this._helpers = await (window as any).loadCardHelpers();
  }

  static get styles(): CSSResult {
    return css``;
  }
}
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by balloob — APPROVED** [review]
This looks good.

In a future PR, we should break up importing the components from the `createXElement` functions. That way we can tweak performance and loading a bit more . -> #4857

File: src/panels/lovelace/custom-card-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { createRowElement as createRowElement_ } from "./create-element/create-row-element";
+import { createCardElement as createCardElement_ } from "./create-element/create-card-element";
+import { createBadgeElement as createBadgeElement_ } from "./create-element/create-badge-element";
+import { createHeaderFooterElement as createHeaderFooterElement_ } from "./create-element/create-header-footer-element";
+import { createHuiElement as createHuiElement_ } from "./create-element/create-hui-element";
```
Comment (bramkragten) [inline src/panels/lovelace/custom-card-helpers.ts]: You can just use export without the `import as` -> `export`
```suggestion
export { createHuiElement } from "./create-element/create-hui-element";
```

**Review by thomasloven — APPROVED** [review]
I like how this is not polluting `window` like anything I would have come up with.

Agree with Bram about deferred exports. That's a neat feature.

**Comment by iantrich** [comment]
thoughts on exposing action handler? That's another thing that keeps changing

**Comment by balloob** [comment]
Makes sense I think. But let's discuss it in a new PR and get this merged so we can do a dev UI release so people can start testing their custom cards against it .

**Comment by bramkragten** [comment]
Does this PR already have documentation?

**Comment by iantrich** [comment]
No

**Pull request merged**